plugin/pkg/up/up.go

```diff
@@ -48,13 +42,19 @@ func (p *Probe) Do(f Func) {
 			if err := f(); err == nil {
 				break
 			}
-			time.Sleep(interval)
 			p.Lock()
 			if p.inprogress == stop {
 				p.Unlock()
 				return
 			}
+			interval := p.expBackoff.NextBackOff()
+			// If exponential backoff has reached the maximum elapsed time, reset it
+			if interval == -1 {
```

Comment:
consider
```go
if interval == backoff.Stop {
```
Shouldn't we stick with max interval (`p.expBackoff.MaxInterval`) in this case?

Reply:
> Shouldn't we stick with max interval (p.expBackoff.MaxInterval) in this case?

IMO yes, that all makes sense - but in this case i was retaining the original logic (and comment), which intentionally resets.

Reply:
ok, I see.
please replace the constant `-1` with `backoff.Stop`